specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/machineLearningServices.json

```diff
@@ -4851,6 +4856,24 @@
       ],
       "type": "object"
     },
+    "ImageMetadata": {
+      "type": "object",
+      "description": "Returns metadata about the operating system image for this compute instance.",
+      "properties": {
+        "currentImageVersion": {
+          "type": "string",
+          "description": "Specifies the current operating system image version this compute instance is running on."
+        },
+        "latestImageVersion": {
+          "type": "string",
+          "description": "Specifies the latest available operating system image version."
+        },
+        "isLatestOsImageVersion": {
+          "type": "boolean",
```

Comment:
ARM guidelines (https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html) strongly suggest AGAINST booleans, saying:
> A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
In this case this variable is set as per currentVersion==latestVersion.

Reply:
Per discussion, we can make this an Enum with [True, False] to represent minimum.

Reply:
As this require extensive backend change and we don't see changing this variable in future. In future we might consider adding another variable that give more extensive information. 
As of now variable name together with boolean value seems good. Moreover this is just in public preview. 
By the time we move to GA we would be in a better state on a need of enum or not.

We therefore plan to keep it as is.

Reply:
Please do consider this when possible, I'm going to sign off for now.